src/main/java/Terry.java

```
@@ -41,14 +46,14 @@ private static void addTask(String description) {
     private static void markTask(int id) {
         if (id < 0 || id >= taskCount) return;
```

Good validation to prevent out-of-bounds access to the tasks array.

---

Suggestion: to print out an error message if an invalid id is given. Otherwise the method silently fails without informing the user.